File: Day 22/CountPrimes.py
#https://leetcode.com/problems/count-primes/
class Solution(object):
    def countPrimes(self, n):
        boundary = int(Math.floor(Math.sqrt(n)))
        counter = 0
        isPrime = []
        for num in range(n):
            isPrime.append(True)
        isPrime[0] = False
        isPrime[1] = False
        for num in range(2,boundary):
            for i in range(len(isPrime)):
                isprime[num*i] = False
        for count in isPrime:
            if(count):
                counter += 1
        return counter

File: Round 2 Preparation/Mathematical/CountPrimes.java
//https://leetcode.com/problems/count-primes/
class Solution {
    public int countPrimes(int n) {
        int primeCount = 0;
        int[] numsList = new int[n];
        for(int index = 2; index < n; index++){
            numsList[index] = 1;
        }
        for(int index = 2; index*index < n; index++){
            if(numsList[index] == 1){
                int squaredValue = (int) Math.pow(index,2);
                while(squaredValue < n){
                    numsList[squaredValue] = 0;
                    squaredValue+=index;
                }
            }
        }
        
        for(int index = 0; index <  n; index++){
            if(numsList[index] == 1){
                primeCount++;
            }
        }
        return primeCount;
        
    }
}